astrogrism/grism_observation.py

```diff
@@ -169,6 +217,9 @@ def _build_geometric_transforms(self, channel=None):
                 if "CCDCHIP" in hdu.header and hdu.header["CCDCHIP"] == channel:
                     hdu_index = i
                     break
+            if hdu_index is None:
```

Comment:
I think you could put this as the `else` to the `for` loop (triggers if the for loop exits normally without a break) and avoid having to initialize `hdu_index` as `None`, but that isn't used terribly often and may not be as readable.

Reply:
Huh, I never knew you could `else` at the end of a `for` loop that way. Seems...gross to me somehow 😆 . I'll think about it